tests/test_circuit.py

```diff
@@ -1,146 +1,171 @@
-import lambeq
 
+# The two lines below will be removed when converting the library to a package.
 current_path = os.path.dirname(os.path.abspath(__file__))
 sys.path.append(current_path + "/../neasqc_wp61/models/quantum/")
-from circuit import *
-
+import circuit as circ
 
 class TestCircuit(unittest.TestCase):
-    
+    """
+    Class for testing Circuit class.
+    """
     @classmethod
-    def setUpClass(cls):
+    def setUpClass(cls)-> None:
         """
-        Set up the class for testing
+        Set up the class for testing.
         """
         torch.manual_seed(args.seed)
         cls.n_qubits = args.n_qubits
         cls.n_layers = args.n_layers
-        cls.axis_encoding = args.axis_encoding
-        if args.observable == 'X':
-            cls.observables = [qml.PauliX for _ in range(args.n_qubits)]
-        elif args.observable == 'Y':
-            cls.observables = [qml.PauliY for _ in range(args.n_qubits)]
-        elif args.observable == 'Z':
-            cls.observables = [qml.PauliZ for _ in range(args.n_qubits)]
-        cls.ansatze = [Sim14, Sim15, StronglyEntangling]
-        cls.circuits = [
+        cls.axis_embedding = args.axis_embedding
+        observables_funcs_dict = {
+            'X' : qml.PauliX, 
+            'Y' : qml.PauliY,
+            'Z' : qml.PauliZ,
+            'H' : qml.Hadamard
+        }
+        cls.observables = {
+            args.qubit_index[i] : observables_funcs_dict[args.observables[i]]
+            for i in range(len(args.qubit_index))
+        }
+        cls.ansatze = [circ.Sim14, circ.Sim15, circ.StronglyEntangling]
+        circuits_exp_value = [
+            ansatz(
+                cls.n_qubits, cls.n_layers,
+                cls.axis_embedding, cls.observables, False
+            ) for ansatz in cls.ansatze
+        ]
+        circuits_probs = [
             ansatz(
                 cls.n_qubits, cls.n_layers,
-                cls.axis_encoding, cls.observables) for ansatz in cls.ansatze
+                cls.axis_embedding, cls.observables, True
+            ) for ansatz in cls.ansatze
         ]
+        cls.circuits = [circuits_exp_value, circuits_probs]
         cls.params = [
             torch.nn.Parameter(
                 torch.randn(
-                    circuit.parameters_shape)) for circuit in cls.circuits
+                    circuit.parameters_shape)) for circuit in cls.circuits[0]
         ]
         cls.input = torch.nn.Parameter(torch.randn(cls.n_qubits))
+        cls.results_circuits = [[], []]
+        for i,c in enumerate(cls.circuits):
+            for j,circuit in enumerate(c):
+                    circuit_function = circuit.build_circuit_function
+                    cls.results_circuits[i].append(
+                        circuit.run_and_measure_circuit(
+                        circuit_function)(cls.input, cls.params[j])
+                    )
 
-
-    def test_circuits_output_correct_type(self):
+    def test_circuits_output_correct_type(self)-> None:
```

Comment:
This name is misleading. When you talk about correctness, you're testing that the VALUE outputted is the CORRECT one. Here it seems you're just testing that it returns a tensor, that's a wholly different story. Your test name should reflect what you're actually testing: that it returns a tensor.